cogs/Voting.py

```diff
@@ -0,0 +1,644 @@
+        """
+        if not opt_id:
+            self.id = randint(100000, 999999)
+        else:
+            self.id = opt_id
+        self.head = head
+        self.body = body
+
+
+class VoteManager:
+    def __init__(self, db_manager):
+        """
+        Manages votes for the bot
+        """
+        self.active_votes = {}
+        self.DBManager = db_manager
+        self.set_up_tables()
+
+    def set_up_tables(self):
+        vote_table = """
+        CREATE TABLE IF NOT EXISTS votes (
+        author_id integer NOT NULL,
+        guild_id integer NOT NULL,
+        title text NOT NULL,
+        chair_id integer,
+        voice_id integer
+        )
+        """
+
+        role_table = """
+        CREATE TABLE IF NOT EXISTS vote_target_roles (
+        vote_author_id integer NOT NULL,
+        role_id integer NOT NULL
+        )"""
+
+        option_table = """
+        CREATE TABLE IF NOT EXISTS vote_options (
+        vote_author_id integer NOT NULL,
+        opt_id integer NOT NULL,
+        option_title text NOT NULL,
+        option_desc text NOT NULL
+        )"""
+
+        delivered_table = """
+        CREATE TABLE IF NOT EXISTS vote_sent (
+        vote_author_id integer NOT NULL,
+        vote_receiver_id integer NOT NULL,
+        vote_receiver_message integer NOT NULL
+        )"""
+
+        self.DBManager.db_execute_commit(vote_table)
+        self.DBManager.db_execute_commit(role_table)
+        self.DBManager.db_execute_commit(option_table)
+        self.DBManager.db_execute_commit(delivered_table)
+
+    def load_from_db(self):
+        existing_votes = self.DBManager.db_execute_select("SELECT * FROM votes")
+        for a_id, g_id, title, chair_id, voice_id in existing_votes:
+            vote = Vote(title, a_id, g_id, self.DBManager)
+            vote.set_chair(chair_id)
+            vote.set_vc(voice_id)
+            self.active_votes[a_id] = vote
+
+            target_roles = self.DBManager.db_execute_select("SELECT * FROM vote_target_roles WHERE vote_author_id=?", (a_id,))
+            if target_roles:
+                for _, r_id in target_roles:
+                    vote.add_role(r_id)
+
+            options = self.DBManager.db_execute_select("SELECT * FROM vote_options WHERE vote_author_id=?", (a_id,))
+            if options:
+                for _, o_id, o_title, o_desc in options:
+                    vote.add_option(Option(o_title, o_desc, opt_id=o_id))
+
+            delivered = self.DBManager.db_execute_select("SELECT * FROM vote_sent WHERE vote_author_id=?", (a_id,))
+            if delivered:
+                for _, rec_id, msg_id in delivered:
+                    vote.register_sent(rec_id, msg_id)
+
+    def get_vote(self, ctx):
+        """
+        Returns a vote from a given discord context
+        :param ctx: discord.Context object from a command
+        :return: Relevant vote object
+        """
+        return self.active_votes[ctx.author.id]
```

Comment:
if ctx.author.id is the only part of ctx that is needed in this (or similarly in other methods) then please only pass that item through (aka author_id instead of ctx) this will make it possible to test this method